### ISSUE TITLE
I propose missed terminology

### ISSUE THREAD

**Issue by a-x-** [bem-site/bem-method#552]
let's start using a few new terms
from: https://gist.github.com/a-x-/78933daf585d236f62831acf078a88d8#complete-formed-bem-blocks-literals

1. **bemslug** `{String}` e.g. `'block__elem_mod_val'`
  slug is a short notation, [//wiki/slug](https://en.wikipedia.org/wiki/Semantic_URL#Slug)
2. **extended-bemslug** `{String}` e.g. `'block_mod_val__elem'`

from https://gist.github.com/a-x-/78933daf585d236f62831acf078a88d8#partial-bem-blocks-declarators-literals-tenorokи:

3. **masked-bemslug** `{String}` e.g. block_* (a mod of block, not elem) etc. ([see more](https://gist.github.com/a-x-/78933daf585d236f62831acf078a88d8#partial-bem-blocks-declarators-literals-tenorokи))

4. bemjson constrains:

  * **bemjson-name** *{Object}* `{block: 'block', elem: 'elem', modName: 'mod', modVal: 'val'}` — single block applicator/matcher declaration
  * **bemjson-decl** *{Object}* `{block: 'block', elem: 'elem', elemMods: {mod:'val', mod2: 'val2'}}` — multiple blocks applicator/matcher declaration
  * **bemjson-def** *{Object}* `{block: 'block', elem: 'elem', elemMods: {mod:'val'}, content: [...],,,}` — general bemjson block definition

**Comment by a-x-** [comment]
bemslug [BNF](https://en.wikipedia.org/wiki/Backus–Naur_Form) like spec: https://gist.github.com/a-x-/78933daf585d236f62831acf078a88d8#bemslug-separators-schemes

**Comment by zxqfox** [comment]
![image](https://cloud.githubusercontent.com/assets/677518/22663354/51c4aa5c-ecbd-11e6-9490-0708acc80bc1.png)

**Comment by a-x-** [comment]
http://s3.hostingkartinok.com/uploads/images/2013/05/5df2ae0589f375d13b8167a1203497ac.jpg

**Comment by zxqfox** [comment]
> 1. bemslug {String} e.g. 'block__elem_mod_val'
> 1. extended-bemslug {String} e.g. 'block_mod_val__elem'

Not sure who needs this and why you need to operate with string representations of BEM entities.

> 3. masked-bemslug {String} e.g. block_* (a mod of block, not elem) etc. (see more)

Even more confused why we need this in methodology.

> 4. bemjson-name {Object} {block: 'block', elem: 'elem', modName: 'mod', modVal: 'val'} — single block applicator/matcher declaration

This is exactly the same what we call [`BEMEntityName`](/bem-sdk/bem-entity-name).

> 5. bemjson-decl {Object} {block: 'block', elem: 'elem', elemMods: {mod:'val', mod2: 'val2'}} — multiple blocks applicator/matcher declaration

This is Node for sure, absolutely not a decl.

> 6. bemjson-def {Object} {block: 'block', elem: 'elem', elemMods: {mod:'val'}, content: [...],,,} — general bemjson block definition

What is the difference between 5 and 6? This is Node again.

**Comment by a-x-** [comment]
ok, let's keep `BEMEntityName`, and we not need `bemjson-name`
so, `BEMEntityName` is for call one entity by name
and `BEMEntityNode` is for call multiple entity 
finnaly, `BEMEntityDef` stands for definition entity, it's not for just call them.

**Comment by a-x-** [comment]
The `string representations of BEM entities` is too long and it's not ready for code term
And yes, i see more cases, when we need term for

**Comment by a-x-** [comment]
I need a bemslug [here](https://github.yandex-team.ru/lego/islands-ci-helpers/pull/231/files) for example. Help me. What are you think, guys?

**Comment by zxqfox** [comment]
`BEMEntityNode`, `BEMEntityDef` → `BEMJSONNode` ?

**Comment by tadatuta** [comment]
I don't think we should extend methodology with new terms if there's just so few cases for them.

**Comment by a-x-** [comment]
I have many cases. l'll gather them here in the future until the term will not stand obvious
